mysql-test/suite/tokudb/r/type_year.result

```diff
@@ -35,7 +35,7 @@ create table t1 (y year);
 insert into t1 values (now());
 select if(y = now(), 1, 0) from t1;
 if(y = now(), 1, 0)
-1
+0
```

Comment:
Why?

Reply:
This works the same as on upstream.
Before: 
In int Arg_comparator::compare_datetime() we compared only years as both items were of type MYSQL_TYPE_YEAR
After:
In the same function we compare year with now() and they are not equal MYSQL_TYPE_YEAR vs MYSQL_TYPE_DATETIME

I didn't investigate it deeper, but it looks like some change around Item_xyz::fix_fields(). I may investigate the exact root cause if it is important, but for me it looks like now it works as it should.